http-custom-mechanism/add-custom-module.cli

```diff
@@ -0,0 +1,3 @@
+# Add the custom module to the server
+
+module add --name=org.jboss.as.quickstart.http_custom_mechanism.custom-http-mechanism --resources=custom-module/target/custom-module.jar --dependencies=org.wildfly.security.elytron
```

Comment:
--allow-nonexistent-resources needs to be added, otherwise it will complain that the resource does not exist.

Reply:
I tried out this quickstart, for some reason I get 404 at [this step ](https://github.com/PrarthonaPaul/quickstart/tree/http-custom-mechanism-enhancement/http-custom-mechanism#test-the-secured-servlet-using-the-custom-mechanism)once I configure the custom mechanism. 
This is the server error when I reload the server: `java.lang.RuntimeException: java.lang.IllegalStateException: There are no mechanisms available from the HttpAuthenticationFactory.`

Reply:
Did you run the script from the quickstart dir? Before the user was instructed to run it from custom-module dir but now that changed to support server provisioning (the path to the jar is related to where the script is invoked)

Reply:
please note that --allow-nonexistent-resources would hide the issue of not finding the jar, which is needed by the custom module mechanism

Reply:
I ran it from the http-custom-mechanism directory for both curl scripts.

Reply:
once CI is done, if it passes, I will try it all from scratch with your updated branch again, and see if I can replicate your issue.